khtools/compare_kmer_content.py

```diff
@@ -195,53 +224,151 @@ def get_comparison_at_index(index, seqlist1, seqlist2,
+
+    if intermediate_csv or intermediate_parquet:
+        # print(id1_sanitized)
+        df = pd.concat(comparision_df_list)
+        if intermediate_csv:
+            df.to_csv(csv)
+        if intermediate_parquet:
+            df.to_parquet(parquet)
+        del df
+    if no_final_concatenation:
+        del comparision_df_list
+        return []
+    else:
+        return comparision_df_list
+
+
+def get_paired_seq_iterator(index, n_background, seqlist1, seqlist2, verbose):
+    """Compare index i for seqlist1 and seqlist2, plus background
+
+    Choose random seqs from seqlist2 for background
+
+    Parameters
+    ----------
+    index : object
+
+
+    """
     pairs_iterator = [(seqlist1[index], seqlist2[index])]
     random_seqlist2 = random.sample(seqlist2, n_background)
     this_index_seqlist1 = [seqlist1[index]] * n_background
     background_pairs = list(zip(this_index_seqlist1, random_seqlist2))
     if verbose:
         print("background_pairs:")
         pprint(background_pairs)
-
-    seq_iterator = list(itertools.chain(*[pairs_iterator, background_pairs]))
+    seq_iterator = list(
+        itertools.chain(*[pairs_iterator, background_pairs]))
     if verbose:
         print("seq_iterator:")
         pprint(seq_iterator)
-    func = partial(compare_args_unpack, ksizes=ksizes, moltype=moltype)
-    comparision_df_list = list(map(func, seq_iterator))
-    notify(
-        "comparison for index {} done in {:.5f} seconds",
-        index,
-        time.time() - startt,
-        end='\r')
-    return comparision_df_list
+    return seq_iterator
 
 
 def compare_all_seqs(seqlist1, seqlist2=None, n_jobs=4, ksizes=KSIZES,
-                     moltype='protein', n_background=100):
-    """
+                     moltype='protein', n_background=100,
+                     paired_seqlists=True, intermediate_csv=False,
+                     intermediate_parquet=False, no_final_concatenation=False):
+    """Compare k-mer content of sequences across k-mer sizes and alphabets
 
     Parameters
     ----------
-    seqlist : list
+    seqlist1 : list
+        List of (id, seq) tuples
+    seqlist2 : list, optional
         List of (id, seq) tuples
+    ksizes : iterable of int
+        K-mer sizes to extract and compare the sequences on
+    moltype : str
+        One of "protein" or "dna" -- for knowing which alphabets to use
+    n_background : int
+        When paired_seqlist is True, how many random background sequences to
+        choose from seqlist2
     n_jobs : int
         Number of jobs for multiprocessing
+    paired_seqlists : bool
+        If True, then seqlist1 and seqlist2 have sequences at the same index
+        that need to be compared, i.e. index 0 across the two
+    intermediate_parquet : bool
+        Write intermediate file of all comparisons at index i to an
+        IO-efficient parquet format
+    intermediate_csv : bool
+        Write intermediate file of all comparisons at index i to an
+        csv format
```

Comment:
for future in docs :raise: ValueError if arg1 is equal to arg2 kind of thing is preferred so user knows what to expect if something is wrong http://queirozf.com/entries/python-docstrings-reference-examples